src/interface/src/styleguide/button/button.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, HostBinding, Input } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { MatIconModule } from '@angular/material/icon';
+
+export type ButtonVariant =
+  | 'primary'
+  | 'ghost'
+  | 'text'
+  | 'negative'
+  | 'positive';
+
+@Component({
+  // eslint-disable-next-line @angular-eslint/component-selector
+  selector: 'button[sg-button]',
```

Comment:
this weird selector is so we can use this component while using the native `button` element vs a custom angular element.